default.nix

```diff
@@ -26,68 +29,69 @@ let
     noTest = pkg: haskell.lib.dontCheck pkg;
 
     hspkgs = haskell.packages.${bootghc};
+    ghc    = haskell.compiler.${bootghc};
 
     ourtexlive =
       nixpkgs.texlive.combine {
         inherit (nixpkgs.texlive) scheme-small collection-xetex fncychap titlesec tabulary varwidth framed capt-of wrapfig needspace dejavu-otf helvetic; };
     fonts = nixpkgs.makeFontsConf { fontDirectories = [ nixpkgs.dejavu_fonts ]; };
     docsPackages = if withDocs then [ python3Packages.sphinx ourtexlive ] else [];
 
-    deps =
+    depsSystem = with stdenv.lib; (
       [ autoconf automake m4
         gmp.dev gmp.out glibcLocales
         ncurses.dev ncurses.out
         perl git file which python3
-        (hspkgs.ghcWithPackages (ps: [ ps.alex ps.happy ]))
         xlibs.lndir  # for source distribution generation
-        cabal-install
         zlib.out
         zlib.dev
       ]
       ++ docsPackages
-      ++ stdenv.lib.optional withLlvm llvm_6
-      ++ stdenv.lib.optional withNuma numactl
-      ++ stdenv.lib.optional withDwarf elfutils
-      ++ stdenv.lib.optional (! stdenv.isDarwin) pxz ;
+      ++ optional  withLlvm            llvm_7
+      ++ optional  withNuma            numactl
+      ++ optional  withDwarf           elfutils
+      ++ optional  (! stdenv.isDarwin) pxz
```

Comment:
I don't like the pointless alignment, now it needs to be maintained if new optionals are added

Reply:
Well, I don't think it needs to be maintained -- I did it as an one-off affair, since so much is touched already..

Reply:
My point is that if things are aligned like this:

```
aaaaa X
bbb   Y
```

then adding an new, longer thing

```
aaaaa X
bbb   Y
cccccccc Z
```

either messes up the alignment or requires extra changes to unrelated lines.

> since so much is touched already..

This actually a reason to do less changes, not more ;)

Reply:
> then adding an new, longer thing
Yes, I understood your point of course -- yes, it messes up alignment -- but in my opinion it still reads better than completely unaligned.

> This is actually a reason to do less changes, not more ;)
This is about batching minor desirable changes that are otherwise not worth doing on their own.

But if you have a different perspective on desirability (i.e. if you don't feel that vaguely table-like layouts are easier to visually scan), then you automatically get different answers on the intermediate questions.

Anyway, I don't want to argue over this, so I'll just cave in : -)

Reply:
Done.

Reply:
Worst case I could have done it myself, but thanks a lot @deepfire-pusher.

I personally can work with either styles so whichever is preferred by other ghc.nix contributors will do.

Reply:
Thank you!